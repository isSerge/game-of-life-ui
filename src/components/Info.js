import React from 'react'
import styled from 'styled-components'

import { dot, Flexed } from './common'

const ColorDot = styled.div`
    ${dot}
    margin-left: 10px;
    background-color: ${({ color }) => color || '#000'};
`

const Wrapper = styled(Flexed)`
    justify-content: space-between;
    width: 500px;
`

const Info = ({ color, generation }) => (
    <Wrapper>
        <Flexed>
            <p>Your color </p>
            <ColorDot color={color} />
        </Flexed>
        <p>Generation: {generation}</p>
    </Wrapper>
)

export default Info
