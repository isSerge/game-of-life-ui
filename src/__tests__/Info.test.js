import React from 'react'
import Info from '../components/Info'
import renderer from 'react-test-renderer'

test('Info matches snapshot', () => {
    const props = {
        color: '#fff',
        generation: 5,
    }
    const component = renderer.create(<Info {...props} />)
    let tree = component.toJSON()

    expect(tree).toMatchSnapshot()
})
