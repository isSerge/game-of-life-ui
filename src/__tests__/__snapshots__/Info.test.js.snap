// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info matches snapshot 1`] = `
<div
  className="sc-bwzfXH sc-bxivhb jAZmXE"
>
  <div
    className="sc-bwzfXH ddDkHF"
  >
    <p>
      Your color 
    </p>
    <div
      className="sc-htpNat foamjo"
      color="#fff"
    />
  </div>
  <p>
    Generation: 
    5
  </p>
</div>
`;
